test/Kestrel.Interop.FunctionalTests/ChromeTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+
+namespace Interop.FunctionalTests
+{
+    [OSSkipCondition(OperatingSystems.MacOSX, SkipReason = "Missing SslStream ALPN support: https://github.com/dotnet/corefx/issues/30492")]
+    [MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win81, SkipReason = "Missing Windows ALPN support: https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation#Support")]
+    [SkipIfChromeUnavailable]
+    [SkipOnJenkins]
+    public class ChromeTests : LoggedTest
+    {
+        private static readonly string _chromeArgs = "--headless --disable-gpu --allow-insecure-localhost --enable-logging --dump-dom --virtual-time-budget=10000";
+
+        [ConditionalTheory]
+        [InlineData("", "Interop HTTP/2 GET")]
+        [InlineData("?TestMethod=POST", "Interop HTTP/2 POST")]
+        public async Task Http2(string requestSuffix, string expectedResponse)
+        {
+            var deploymentParameters = new DeploymentParameters()
+            {
+                ApplicationPath = Path.Combine(TestResources.SolutionePath, "test", "Kestrel.Interop.TestSites"),
+                ServerType = ServerType.Kestrel,
+                TargetFramework = Tfm.NetCoreApp22,
+                RuntimeArchitecture = RuntimeArchitecture.x64
+            };
+
+            using (var deployer = new SelfHostDeployer(deploymentParameters, LoggerFactory))
```

Comment:
Why do you need to deploy? Can't you run Kestrel in proc?

Reply:
It's the pattern I've found in ServerTests.

Reply:
The deployers are only really needed for IIS, and because we wanted to test the various publish options. None of that is relevant here, in-proc tests will be much simpler.